tests/flows/test_vesting.py

```diff
@@ -171,3 +172,150 @@ def test_main(
+    ocean_contract.approve(
+        get_address_of_type(config, "Router"), initial_ocean_liq, publisher_wallet
+    )
+    vested_blocks = factory_router.get_min_vesting_period()
+    percentages = [0.01, 0.1, 1]
+    checkpoint_blocks = list(map(lambda p: int((vested_blocks * p) / 100), percentages))
+
+    vesting_amount = initial_ocean_liq // 100 * 9
+
+    tx = erc20_token.deploy_pool(
+        rate=to_wei(1),
+        basetoken_decimals=ocean_contract.decimals(),
+        vesting_amount=vesting_amount,
+        vested_blocks=vested_blocks,
+        initial_liq=initial_ocean_liq,
+        lp_swap_fee=to_wei("0.003"),
+        market_swap_fee=to_wei("0.001"),
+        ss_contract=get_address_of_type(config, "Staking"),
+        basetoken_address=ocean_contract.address,
+        basetoken_sender=publisher_wallet.address,
+        publisher_address=publisher_wallet.address,
+        market_fee_collector=get_address_of_type(config, "OPFCommunityFeeCollector"),
+        pool_template_address=get_address_of_type(config, "poolTemplate"),
+        from_wallet=publisher_wallet,
+    )
+    tx_receipt = web3.eth.wait_for_transaction_receipt(tx)
+    pool_event = erc20_token.get_event_log(
+        ERC721FactoryContract.EVENT_NEW_POOL,
+        tx_receipt.blockNumber,
+        web3.eth.block_number,
+        None,
+    )
+
+    assert pool_event[0].event == "NewPool", "Cannot find NewPool event"
+    assert pool_event[0].args.ssContract == get_address_of_type(config, "Staking")
+    bpool_address = pool_event[0].args.poolAddress
+    bpool = BPool(web3, bpool_address)
+    assert bpool.is_finalized() is True
+
+    side_staking = SideStaking(
+        web3=web3, address=get_address_of_type(config, "Staking")
+    )
+    vesting_created_event = side_staking.get_event_log(
+        SideStaking.EVENT_VESTING_CREATED,
+        tx_receipt.blockNumber,
+        web3.eth.block_number,
+        None,
+    )
+    assert vesting_created_event, "Cannot find event VestingCreated."
+    assert vesting_created_event[0].args.datatokenAddress == erc20_token.address
+    assert (
+        vesting_created_event[0].args.publisherAddress
+        == erc20_token.get_payment_collector()
+    )
+    assert (
+        vesting_created_event[0].args.vestingEndBlock
+        == vested_blocks + tx_receipt.blockNumber
+    )
+    assert vesting_created_event[0].args.totalVestingAmount == vesting_amount
+
+    # check if vesting is correct
+    dt_balance_before = erc20_token.balanceOf(erc20_token.get_payment_collector())
+    available_vesting_before = side_staking.get_available_vesting(erc20_token.address)
+    assert available_vesting_before == 0
+
+    # advance checkpoint_block blocks to see if available vesting increased
+    for block in range(checkpoint_blocks[-1]):
+        # send dummy transactions to increase block count (not part of the actual functionality)
+        send_ether(publisher_wallet, ZERO_ADDRESS, 1)
+        if block in checkpoint_blocks:
```

Comment:
yes, but as I said previously, do not add a layer of imbrication.
if the block is not a checkpoint block, `continue`
then go ahead with the checkpoint block logic (lines 298-321, but unindented by 1 measure)

Reply:
Ok, sure. I will change the condition logic. Also I will add the functionality for claiming the datatokens as well.